test/TEST-03-JOBS/test-jobs.sh

```diff
@@ -10,44 +10,44 @@ while ! grep 'sleep\.service.*running' /root/list-jobs.txt; do
     systemctl list-jobs > /root/list-jobs.txt
 done
 
-grep 'hello\.service.*waiting' /root/list-jobs.txt || exit 1
+grep 'hello\.service.*waiting' /root/list-jobs.txt
 
 # This is supposed to finish quickly, not wait for sleep to finish.
 START_SEC=$(date -u '+%s')
 systemctl start --job-mode=ignore-dependencies hello
 END_SEC=$(date -u '+%s')
 ELAPSED=$(($END_SEC-$START_SEC))
 
-[ "$ELAPSED" -lt 3 ] || exit 1
+[ "$ELAPSED" -lt 3 ]
 
 # sleep should still be running, hello not.
 systemctl list-jobs > /root/list-jobs.txt
-grep 'sleep\.service.*running' /root/list-jobs.txt || exit 1
-grep 'hello\.service' /root/list-jobs.txt && exit 1
-systemctl stop sleep.service hello-after-sleep.target || exit 1
+grep 'sleep\.service.*running' /root/list-jobs.txt
+! grep 'hello\.service' /root/list-jobs.txt
```

Comment:
According to the `bash` man page, the shell does not exit if the return value of a command is inverted with `!`, so it seems that `! cmd` cannot be used instead of `cmd && exit 1`.